app/code/Magento/InventorySales/Model/IsProductSalableCondition/IsSalableLegacyStockItemIsInStock.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\IsProductSalableCondition;
+
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\CatalogInventory\Model\Spi\StockRegistryProviderInterface;
+use Magento\InventoryCatalogApi\Model\GetProductIdsBySkusInterface;
+use Magento\InventoryConfiguration\Model\IsSourceItemManagementAllowedForSku;
+
+/**
+ * @inheritdoc
+ */
+class IsSalableLegacyStockItemIsInStock implements IsProductSalableInterface
+{
+    /**
+     * @var IsSourceItemManagementAllowedForSku
+     */
+    private $isSourceItemManagementAllowedForSku;
+
+    /**
+     * @var StockRegistryProviderInterface
+     */
+    private $stockRegistryProvider;
+
+    /**
+     * @var StockConfigurationInterface
+     */
+    private $stockConfiguration;
+
+    /**
+     * @var GetProductIdsBySkusInterface
+     */
+    private $getProductIdsBySkus;
+
+    /**
+     * @param IsSourceItemManagementAllowedForSku $isSourceItemManagementAllowedForSku
+     * @param StockRegistryProviderInterface $stockRegistryProvider
+     * @param StockConfigurationInterface $stockConfiguration
+     * @param GetProductIdsBySkusInterface $getProductIdsBySkus
+     */
+    public function __construct(
+        IsSourceItemManagementAllowedForSku $isSourceItemManagementAllowedForSku,
+        StockRegistryProviderInterface $stockRegistryProvider,
+        StockConfigurationInterface $stockConfiguration,
+        GetProductIdsBySkusInterface $getProductIdsBySkus
+    ) {
+        $this->isSourceItemManagementAllowedForSku = $isSourceItemManagementAllowedForSku;
+        $this->stockRegistryProvider = $stockRegistryProvider;
+        $this->stockConfiguration = $stockConfiguration;
+        $this->getProductIdsBySkus = $getProductIdsBySkus;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, int $stockId): bool
+    {
+        if ($this->isSourceItemManagementAllowedForSku->execute($sku)) {
+            return true;
+        }
```

Comment:
why do we have this check? 
```
        if ($this->isSourceItemManagementAllowedForSku->execute($sku)) {
            return true;
        }
```

this service just checks whether product type is allowed Source Item management